migrator/migrations/m_168_to_m_169_postgres_remove_clustercve_permission/permissionsetpostgresstore/postgres_plugin.go

```diff
@@ -0,0 +1,283 @@
+package postgres
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/pkg/errors"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/logging"
+	ops "github.com/stackrox/rox/pkg/metrics"
+	"github.com/stackrox/rox/pkg/postgres/pgutils"
+	pkgSchema "github.com/stackrox/rox/pkg/postgres/schema"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/search/postgres"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+// This file is a partial copy of central/role/store/permissionset/postgres/store.go
+// in the state it had when the migration was written.
+// Only the relevant functions (Walk, DeleteMany and UpsertMany) are kept.
+// The kept functions are stripped from the scoped access control checks.
+
+const (
+	baseTable = "permission_sets"
+
+	batchAfter = 100
+
+	// using copyFrom, we may not even want to batch.  It would probably be simpler
+	// to deal with failures if we just sent it all.  Something to think about as we
+	// proceed and move into more e2e and larger performance testing
+	batchSize = 10000
+
+	cursorBatchSize = 50
+	deleteBatchSize = 5000
+)
+
+var (
+	log    = logging.LoggerForModule()
+	schema = pkgSchema.PermissionSetsSchema
+)
+
+// Store is the interface for interactions with the database storage
+type Store interface {
+	DeleteMany(ctx context.Context, ids []string) error
+	UpsertMany(ctx context.Context, objs []*storage.PermissionSet) error
+	Walk(ctx context.Context, fn func(obj *storage.PermissionSet) error) error
+}
+
+type storeImpl struct {
+	db    *pgxpool.Pool
+	mutex sync.Mutex
+}
+
+// New returns a new Store instance using the provided sql instance.
+func New(db *pgxpool.Pool) Store {
+	return &storeImpl{
+		db: db,
+	}
+}
+
+func insertIntoPermissionSets(ctx context.Context, batch *pgx.Batch, obj *storage.PermissionSet) error {
+
+	serialized, marshalErr := obj.Marshal()
+	if marshalErr != nil {
+		return marshalErr
+	}
+
+	values := []interface{}{
+		// parent primary keys start
+		pgutils.NilOrUUID(obj.GetId()),
+		obj.GetName(),
+		serialized,
+	}
+
+	finalStr := "INSERT INTO permission_sets (Id, Name, serialized) VALUES($1, $2, $3) ON CONFLICT(Id) DO UPDATE SET Id = EXCLUDED.Id, Name = EXCLUDED.Name, serialized = EXCLUDED.serialized"
+	batch.Queue(finalStr, values...)
+
+	return nil
+}
+
+func (s *storeImpl) copyFromPermissionSets(ctx context.Context, tx pgx.Tx, objs ...*storage.PermissionSet) error {
+
+	inputRows := [][]interface{}{}
+
+	var err error
+
+	// This is a copy so first we must delete the rows and re-add them
+	// Which is essentially the desired behaviour of an upsert.
+	var deletes []string
+
+	copyCols := []string{
+
+		"id",
+
```

Comment:
Not sure how these empty lines ended up in the code but they are present in `role/store/permissionset/postgres/store.go` as well, so NNTR.